handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -1246,10 +1238,14 @@ private void closeOutboundAndChannel(
         outboundClosed = true;
         engine.closeOutbound();
 
-        ChannelPromise closeNotifyFuture = ctx.newPromise();
-        write(ctx, Unpooled.EMPTY_BUFFER, closeNotifyFuture);
+        ChannelPromise closeNotifyPromise = ctx.newPromise();
+        flush(ctx, closeNotifyPromise);
+        safeClose(ctx, closeNotifyPromise, promise);
```

Comment:
i seem to remember a recent PR where closing was put into a finally block ... or we excplicity called close if an exception occurred ... does this need PR need to be rebased or do we need to do the same thing here?